source/common/router/BUILD

```diff
@@ -22,6 +22,7 @@ envoy_cc_library(
         "//include/envoy/http:header_map_interface",
         "//include/envoy/router:router_interface",
         "//include/envoy/runtime:runtime_interface",
+        "//include/envoy/server:filter_config_interface",
```

Comment:
In general we try to not have common depend on server, though this isn't really enforced explicitly. I would add a TODO here and in code to figure out a better way to break this cycle.

Reply:
sg. i imagine this could be done with bazel visibility rules?

Reply:
I need to think about it. Will look at it as part of repo reorg work.